AsyncDisplayKit/Layout/ASDimension.mm

```diff
@@ -11,66 +11,243 @@
+  }
+  
+  // Handle points
+  NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
+  if (pointsStringLocation != NSNotFound) {
+    // Check if points is at the end and remove it
+    if (pointsStringLocation == (dimension.length-2)) {
+      dimension = [dimension substringToIndex:(dimension.length-2)];
+      return ASDimensionMake(ASDimensionTypePoints, dimension.floatValue);
+    }
+  }
+  
+  // Handle fraction
+  NSUInteger percentStringLocation = [dimension rangeOfString:@"%"].location;
+  if (percentStringLocation != NSNotFound) {
+    // Check if percent is at the end and remove it
+    if (percentStringLocation == (dimension.length-1)) {
+      dimension = [dimension substringToIndex:(dimension.length-1)];
+      return ASDimensionMake(ASDimensionTypeFraction, dimension.floatValue);
+    }
+  }
+
+  // Assert as parsing went wrong
+  ASDisplayNodeCAssert(NO, @"Parsing dimension failed");
+  return ASDimensionAuto;
+}
+
+ASDimension ASDimensionMakeWithPoints(CGFloat points)
 {
   ASDisplayNodeCAssertPositiveReal(@"Points", points);
-  return ASRelativeDimensionMake(ASRelativeDimensionTypePoints, points);
+  return ASDimensionMake(ASDimensionTypePoints, points);
 }
 
-ASRelativeDimension ASRelativeDimensionMakeWithFraction(CGFloat fraction)
+ASDimension ASDimensionMakeWithFraction(CGFloat fraction)
 {
-  // ASDisplayNodeCAssert( 0 <= fraction && fraction <= 1.0, @"ASRelativeDimension fraction value (%f) must be between 0 and 1.", fraction);
-  return ASRelativeDimensionMake(ASRelativeDimensionTypeFraction, fraction);
+  // ASDisplayNodeCAssert( 0 <= fraction && fraction <= 1.0, @"ASDimension fraction value (%f) must be between 0 and 1.", fraction);
+  return ASDimensionMake(ASDimensionTypeFraction, fraction);
 }
 
-ASRelativeDimension ASRelativeDimensionCopy(ASRelativeDimension aDimension)
+ASDimension ASDimensionCopy(ASDimension aDimension)
 {
-  return ASRelativeDimensionMake(aDimension.type, aDimension.value);
+  return ASDimensionMake(aDimension.type, aDimension.value);
 }
 
-BOOL ASRelativeDimensionEqualToRelativeDimension(ASRelativeDimension lhs, ASRelativeDimension rhs)
+BOOL ASDimensionEqualToDimension(ASDimension lhs, ASDimension rhs)
 {
   return lhs.type == rhs.type && lhs.value == rhs.value;
 }
 
-NSString *NSStringFromASRelativeDimension(ASRelativeDimension dimension)
+NSString *NSStringFromASDimension(ASDimension dimension)
 {
   switch (dimension.type) {
-    case ASRelativeDimensionTypePoints:
+    case ASDimensionTypePoints:
       return [NSString stringWithFormat:@"%.0fpt", dimension.value];
-    case ASRelativeDimensionTypeFraction:
+    case ASDimensionTypeFraction:
       return [NSString stringWithFormat:@"%.0f%%", dimension.value * 100.0];
+    case ASDimensionTypeAuto:
+      return @"Auto";
   }
 }
 
-CGFloat ASRelativeDimensionResolve(ASRelativeDimension dimension, CGFloat parent)
+CGFloat ASDimensionResolve(ASDimension dimension, CGFloat autoSize, CGFloat parent)
```

Comment:
Optional: Consider a more descriptive name here. This method name isn't particular clear on what it does exactly.